src/cdi_analytic/Compound_Analytic_MultigroupOpacity.cc

```diff
@@ -120,28 +124,32 @@ nGray_Analytic_MultigroupOpacity::nGray_Analytic_MultigroupOpacity(
  * Given a scalar temperature and density, return the group opacities
  * (vector<double>) for the reaction type specified by the constructor.  The
  * analytic opacity model is specified in the constructor
- * (nGray_Analytic_MultigroupOpacity()).
+ * (Compound_Analytic_MultigroupOpacity()).
  *
  * \param temperature material temperature in keV
  * \param density material density in g/cm^3
  * \return group opacities (coefficients) in cm^2/g
  *
  */
-nGray_Analytic_MultigroupOpacity::sf_double
-nGray_Analytic_MultigroupOpacity::getOpacity(double temperature,
-                                             double density) const {
+Compound_Analytic_MultigroupOpacity::sf_double
+Compound_Analytic_MultigroupOpacity::getOpacity(double temperature,
+                                                double density) const {
   Require(temperature >= 0.0);
   Require(density >= 0.0);
 
   // return opacities
   sf_double opacities(group_models.size(), 0.0);
 
   // loop through groups and get opacities
+  auto nu = getGroupBoundaries();
+  Check(nu.size() == group_models.size() + 1);
+
   for (size_t i = 0; i < opacities.size(); ++i) {
     Check(group_models[i]);
 
     // assign the opacity based on the group model
-    opacities[i] = group_models[i]->calculate_opacity(temperature, density);
+    opacities[i] = group_models[i]->calculate_opacity(temperature, density,
+                                                      nu[i], nu[i + 1]);
```

Comment:
Just to be sure I understand -- this update will cause a change in behavior if the group_model is Polynomial_Analytic_Opacity_Model, correct?

Reply:
Yes. This is the change that brings the behavior into consistency with the ODF version.

Reply:
Excellent, thanks!